docs/dev/table/hive/hive_read_write.md

```diff
@@ -22,119 +22,199 @@ specific language governing permissions and limitations
+    </tr>
+    <tr>
+        <td><h5>streaming-source.consume-start-offset</h5></td>
+        <td style="word-wrap: break-word;">1970-00-00</td>
+        <td>String</td>
+        <td>Start offset for streaming consuming. How to parse and compare offsets depends on your order. For create-time and partition-time, should be a timestamp string (yyyy-[m]m-[d]d [hh:mm:ss]). For partition-time, will use partition time extractor to extract time from partition.</td>
+    </tr>
+  </tbody>
+</table>
+
+[SQL Hints]({% link dev/table/sql/hints.md %}) can be used to apply configurations to a Hive table
+without changing its definition in the Hive metastore.
+
+{% highlight sql %}
+
+SELECT * 
+FROM hive_table 
+/*+ OPTIONS('streaming-source.enable'='true', 'streaming-source.consume-start-offset'='2020-05-20') */;
 
-{% highlight bash %}
-hive> show databases;
-OK
-default
-Time taken: 0.841 seconds, Fetched: 1 row(s)
-
-hive> show tables;
-OK
-Time taken: 0.087 seconds
-
-hive> CREATE TABLE mytable(name string, value double);
-OK
-Time taken: 0.127 seconds
-
-hive> SELECT * FROM mytable;
-OK
-Tom   4.72
-John  8.0
-Tom   24.2
-Bob   3.14
-Bob   4.72
-Tom   34.9
-Mary  4.79
-Tiff  2.72
-Bill  4.33
-Mary  77.7
-Time taken: 0.097 seconds, Fetched: 10 row(s)
 {% endhighlight %}
 
-With the data ready your can connect to Hive [connect to an existing Hive installation]({{ site.baseurl }}/dev/table/hive/#connecting-to-hive) and begin querying. 
+**Notes**
 
-{% highlight bash %}
+- Monitor strategy is to scan all directories/files currently in the location path. Many partitions may cause performance degradation.
+- Streaming reads for non-partitioned tables requires that each file be written atomically into the target directory.
+- Streaming reading for partitioned tables requires that each partition should be added atomically in the view of hive metastore. If not, new data added to an existing partition will be consumed.
+- Streaming reads do not support watermark grammar in Flink DDL. These tables cannot be used for window operators.
 
-Flink SQL> show catalogs;
-myhive
-default_catalog
+## Reading Hive Views
 
-# ------ Set the current catalog to be 'myhive' catalog if you haven't set it in the yaml file ------
+Flink is able to read from Hive defined views, but some limitations apply:
 
-Flink SQL> use catalog myhive;
+1) The Hive catalog must be set as the current catalog before you can query the view. 
+This can be done by either `tableEnv.useCatalog(...)` in Table API or `USE CATALOG ...` in SQL Client.
 
-# ------ See all registered database in catalog 'mytable' ------
+2) Hive and Flink SQL have different syntax, e.g. different reserved keywords and literals.
+Make sure the view’s query is compatible with Flink grammar.
 
-Flink SQL> show databases;
-default
+### Temporal Table Join
```

Comment:
I wonder whether we should list all the source tables that can be used as temporal tables (maybe in [this page](https://ci.apache.org/projects/flink/flink-docs-release-1.11/dev/table/streaming/temporal_tables.html#temporal-table)). And then we can put a cross link here. It's annoying for users to have to check each connector doc to decide whether a table can be used in temporal join. And maybe users don't even expect to find such information in the "read/write" page.

Reply:
There's an open PR to expand temporal table join documentation, why don't you comment there #14152